bin/pyrdp-mitm.py

```diff
@@ -7,53 +7,75 @@
 import logging
 import logging.handlers
 import os
+import random
 import sys
-from typing import Optional
+from pathlib import Path
 
 import appdirs
 import names
 from twisted.internet import reactor
 from twisted.internet.protocol import ServerFactory
 
-from pyrdp.core import getLoggerPassFilters
-from pyrdp.logging import JSONFormatter, log, LOGGER_NAMES, SensorFilter
-from pyrdp.mitm import MITMServer
+from pyrdp.logging import JSONFormatter, log, LOGGER_NAMES, SessionLogger, VariableFormatter
+from pyrdp.mitm import MITMConfig, RDPMITM
 
 
 class MITMServerFactory(ServerFactory):
-    def __init__(self, targetHost: str, targetPort: int, privateKeyFileName: str, certificateFileName: str,
-                 destination_ip: str, destination_port: int, username: Optional[str], password: Optional[str]):
+    """
+    Server factory for the RDP man-in-the-middle that generates a unique session ID for every connection.
+    """
+
+    def __init__(self, config: MITMConfig):
         """
-        :param targetHost: The IP that points to the RDP server
-        :param targetPort: The port that points to the RDP server
-        :param privateKeyFileName: The private key to use for SSL
-        :param certificateFileName: The certificate to use for SSL
-        :param destination_ip: The IP to which send RDP traffic (for live player).
-        :param destination_port: The port to which send RDP traffic (for live player).
-        :param username: The replacement username to use to connect users instead of the one they provided.
-        :param password: The replacement password to use to connect users instead of the one they provided.
+        :param config: the MITM configuration
         """
-        self.password = password
-        self.username = username
-        self.targetHost = targetHost
-        self.targetPort = targetPort
-        self.privateKeyFileName = privateKeyFileName
-        self.certificateFileName = certificateFileName
-        self.destination_ip = destination_ip
-        self.destination_port = destination_port
+        self.config = config
 
     def buildProtocol(self, addr):
-        server = MITMServer(names.get_first_name(), self.targetHost, self.targetPort, self.certificateFileName,
-                            self.privateKeyFileName, self.destination_ip, self.destination_port,
-                            self.username, self.password)
-        return server.getProtocol()
+        sessionID = f"{names.get_first_name()}{random.randrange(100000,999999)}"
+        logger = logging.getLogger(LOGGER_NAMES.MITM_CONNECTIONS)
+        logger = SessionLogger(logger, sessionID)
+        mitm = RDPMITM(logger, self.config)
+
+        return mitm.getProtocol()
+
+
+def prepareLoggers(logLevel: int, sensorID: str, outDir: Path):
+    logDir = outDir / "logs"
+    logDir.mkdir(exist_ok = True)
+
+    formatter = VariableFormatter("[{asctime}] - {levelname} - {sessionID} - {name} - {message}", style = "{", defaultVariables = {
```

Comment:
So now the sessionID is not a part of the logger name anymore, right?

Reply:
Nope